experiments/AMIP/moist_mpi_earth/push_pull.jl

```diff
@@ -1,180 +1,152 @@
+function ocean_pull!(cs)
+    ocean_sim = cs.model_sims.ocn
+    csf = cs.fields
+    @. ocean_sim.integrator.p.F_aero = csf.F_A
+    @. ocean_sim.integrator.p.F_rad = csf.F_R
 end
 
 """
-   ice_pull!(slab_ice_sim, F_A, F_R)
-
-Updates the slab_ice_sim cache state in place with the current values of F_A and F_R.
-
+   ice_pull!(cs)
+Updates the ice_sim cache state in place with the current values of F_A and F_R.
 In the current version, the sea ice has a prescribed thickness, and we assume that it is not
 sublimating. That contribution has been zeroed out in the atmos fluxes.
 """
-function ice_pull!(slab_ice_sim, F_A, F_R)
-    @. slab_ice_sim.integrator.p.Ya.F_aero = F_A
-    @. slab_ice_sim.integrator.p.Ya.F_rad = F_R
+function ice_pull!(cs)
+    ice_sim = cs.model_sims.ice
+    csf = cs.fields
+    @. ice_sim.integrator.p.Ya.F_aero = csf.F_A
+    @. ice_sim.integrator.p.Ya.F_rad = csf.F_R
 end
 
 """
-   atmos_pull!(atmos_sim,
-               slab_ice_sim,
-               slab_sim,
-               slab_ocean_sim,
-               boundary_space,
-               prescribed_sst,
-               z0m_S,
-               z0b_S,
-               T_S,
-               ρ_sfc,
-               q_sfc,
-               ocean_params,
-               SST,
-               land_sea_mask
-              )
+   atmos_pull!(cs)
+              
 Creates the surface fields for temperature, roughness length, albedo, and specific humidity; computes
 turbulent surface fluxes; updates the atmosphere boundary flux cache variables in place; updates the
 RRTMGP cache variables in place.
 """
-function atmos_pull!(
-    atmos_sim,
-    slab_ice_sim,
-    slab_sim,
-    slab_ocean_sim,
-    boundary_space,
-    prescribed_sst,
-    z0m_S,
-    z0b_S,
-    T_S,
-    ρ_sfc,
-    q_sfc,
-    ocean_params,
-    SST,
-    land_sea_mask,
-)
+function atmos_pull!(cs)
+
+    atmos_sim = cs.model_sims.atm
+    slab_sim = cs.model_sims.lnd
+    ocean_sim = cs.model_sims.ocn
+    ice_sim = cs.model_sims.ice
+
+    csf = cs.fields
+    T_sfc_cpl = csf.T_S
+    z0m_cpl = csf.z0m_S
+    z0b_cpl = csf.z0b_S
+    ρ_sfc_cpl = csf.ρ_sfc
+    q_sfc_cpl = csf.q_sfc
+    albedo_sfc_cpl = csf.albedo
+
     thermo_params = CAP.thermodynamics_params(atmos_sim.integrator.p.params)
 
-    univ_mask = parent(land_sea_mask) .- parent(slab_ice_sim.integrator.p.Ya.ice_mask .* FT(2))
     T_land = get_land_temp(slab_sim)
     z0m_land, z0b_land = get_land_roughness(slab_sim)
+    T_ocean = ocean_sim.integrator.u.T_sfc
+    z0m_ocean = ocean_sim.integrator.p.params.z0m
+    z0b_ocean = ocean_sim.integrator.p.params.z0b
+    α_ocean = ocean_sim.integrator.p.params.α
+    T_ice = ice_sim.integrator.u.T_sfc
+    ice_mask = ice_sim.integrator.p.Ya.ice_mask
+    z0m_ice = ice_sim.integrator.p.params.z0m
+    z0b_ice = ice_sim.integrator.p.params.z0b
+
+    land_mask = cs.mask
+
+    # mask of all models (1 = land, 0 = ocean, -2 = sea ice)
+    univ_mask = parent(land_mask) .- parent(ice_mask .* FT(2))
```

Comment:
get FT from coupler sim object, right now it is looking up in global here I think